dex/wait/queue.go

```diff
@@ -253,14 +248,37 @@ func (q *TaperingTickerQueue) Run(ctx context.Context) {
 			sort.Slice(waiters, func(i, j int) bool {
 				return waiters[i].nextTick.Before(waiters[j].nextTick) // ascending, next tick first
 			})
-			// NOTE: timer leaked until it fires - consider NewTimer and Stop here
 
 		case <-ctx.Done():
+			maybeStop(timer)
 			for _, w := range waiters {
 				w.ExpireFunc() // early, but still ending prior to DontTryAgain
 			}
 			return
 		}
 	}
+}
 
+func nextTick(
+	nextTick int,
+	slowestInterval, fastestInterval time.Duration,
+	now, expiration time.Time,
+) time.Time {
+	var nextTickTime time.Time
+	switch {
+	case nextTick < fullSpeedTicks:
+		nextTickTime = now.Add(fastestInterval)
+	case nextTick < fullyTapered: // ramp up the interval
+		prog := float64(nextTick+1-fullSpeedTicks) / (fullyTapered - fullSpeedTicks)
```

Comment:
minor change that makes this behavior slightly more precise with respect to the comment on:
```go
    // fullSpeedTicks is the number of attempts that will be made with the
	// configured fastestInterval delay. After fullSpeedTicks, the retry speed
	// will be tapered off.
	fullSpeedTicks = 3
```
since `tick` starts at 0, not 1